### ISSUE TITLE
Doesn't work if system time is changed

### ISSUE THREAD

**Issue by jduranf** [jsln/pps-gen-gpio#4]
If system time is changed forwards, the PPS generator stars sending `We are late this time...` continuously. This can be fixed by replacing `ts_expire_req` by `ts_expire_real` in this [line](https://github.com/jsln/pps-gen-gpio/blob/master/pps_gen_gpio.c#L136).

**Comment by jsln** [comment]
Thanks for the input, I guess if system time is changed forwards, you see the "We are late this time..." for a while before we catch up, right? I will keep this into account on the next update.

**Comment by jduranf** [comment]
Yes. Take in account that my suggestion fix the issue when the time is changed forwards, but not when the time is changed backwards.
Also, another suggestion is to change the "We are late this time..." from error message to debug message. When the module is working fine, we see this message every several seconds.

**Comment by Roozter** [comment]
Will I have issues if I use this generator while my system is being synchronized with an external GPS?

**Comment by jsln** [comment]
> Will I have issues if I use this generator while my system is being synchronized with an external GPS?

I used it this way and I did not notice any issue related to the GPS capture.

**Comment by jduranf** [comment]
Hi @jsln! Seems this issue isn't still fixed. If it's fine for you, I can create a PR to fix it.

**Comment by jsln** [comment]
> Hi [@jsln](https://github.com/jsln)! Seems this issue isn't still fixed. If it's fine for you, I can create a PR to fix it.

Hi @jduranf , this issue has been there for ages, feel free to create a PR, thanks!

**Comment by jsln** [comment]
Thanks a lot for your PR, Joan, I am closing your issue.
